입출금 현황 화면 레이아웃 작성

donggeonoh opened the pull request:
구현 사항

UICollectionView를 이용하여 입출금 현황 화면 구성
UICollectionViewLayout의 ListConfiguration을 통해 리스트로 구현
UICollectionViewDiffableDataSource를 이용하여 모델 관리

참고로 MockData 타입과 모델 관련해서 임시적으로 추가했는데요. (테스트를 위해)
이부분 나중에 API 연결했을 때 삭제 해주시면 됩니다.

<img width="334" alt="Screen Shot 2022-02-25 at 11 51 27 PM" src="https://user-images.githubusercontent.com/18098363/155735909-476cb7c7-84a7-4845-b5cb-8173d2fd770a.png">

donggeonoh committed:
feat: 전체적인 입출금 현황 화면 레이아웃 작성 # 21

UICollectionView를 이용하여 입출금 현황 화면 구성
UICollectionViewLayout의 ListConfiguration을 통해 리스트로 구현
UICollectionViewDiffableDataSource를 이용하여 모델 관리

z3rosmith commented on BithumbYagomAcamedy/Controller/DepositWithdrawalStatusViewController.swift:
```diff
@@ -0,0 +1,76 @@
+        configureMockData()
+    }
+    
+    // - MARK: Initialization
```
Initialization은 ViewController의 init과 헷갈릴 여지가 있어서 밑의 configure~메서드와 같이 // MARK: - Configuration으로 하는건 어떠세요??

donggeonoh replied:
아 그렇군요 의견 반영하겠습니다~

z3rosmith commented on BithumbYagomAcamedy/Controller/DepositWithdrawalStatusViewController.swift:
```diff
@@ -0,0 +1,76 @@
+    
+    private func configureDiffableDataSource() {
+        let cellRegistration = UICollectionView.CellRegistration<DepositWithdrawalCollectionViewCell, MockData>(cellNib: UINib(nibName: depositWithdrawalCollectionViewCellNibName, bundle: nil)) { cell, indexPath, item in
+            cell.update(item)
```
오 이것처럼 cell update하는것 좋네요 저도 참고하겠습니다

donggeonoh replied:
도움이 됐다니 기쁘네요 ㅎㅎ

donggeonoh committed:
feat: Data의 UUID 적용 및 MARK 주석 수정 #21

donggeonoh committed:
feat: 입출금 현황 상태 색깔 Asset에 추가 및 Cell에 적용 #21

donggeonoh committed:
Merge branch 'develop' into feat/21/implement-deposit-and-withdrawal-view

HJEHA commented on BithumbYagomAcamedy/Controller/DepositWithdrawalStatusViewController.swift:
```diff
@@ -0,0 +1,98 @@
+        let symbol: String
+        let depositStatus: Bool
+        let withdrawalStatus: Bool
+        let uuid: UUID
```
UUID는 기본값을 넣어줘도 괜찮을 것 같습니다:)

donggeonoh replied:
의견 감사합니다~ 기본 값을 넣어줘도 좋겠네요.

HJEHA commented on BithumbYagomAcamedy/View/DepositWithdrawalStatusView/DepositWithdrawalCollectionViewCell.swift:
```diff
@@ -7,11 +7,32 @@
+        withdrawalStatusView.layer.cornerRadius = withdrawalStatusView.frame.width / 2
+    }
+    
+    func update(_ data: DepositWithdrawalStatusViewController.MockData) {
```
현재는 테스트을 위해서 매개변수에 MockData 타입이 들어간 것 같은데
추후 개발에서는 구체타입 보다 프로토콜 타입을 받아서 처리하면 재사용에 용의할 것 같습니다:)

donggeonoh replied:
좋은 것 같습니다. 필요한 데이터만 모델에서 넘겨주면 되니까요!

HJEHA reviewed (commented):
코멘트 확인 부탁드립니다:)

donggeonoh committed:
feat: UUID 기본 값 초기화 및 view model protocol 추가

UUID는 기본으로 생성하도록 MockData를 만듦
입출금 현황 뷰 셀의 경우 update 시 protocol 을 하나 선언하여
protocol을 만족하는 타입이 넘어오도록 하여 의존성을 낮춤
